frontend/components/common/CompatibleDevice.tsx

```diff
@@ -26,102 +28,134 @@ export function CompatibleDevice({
       maxModelLines > 0 && variants.length > maxModelLines
          ? [variants.slice(0, maxModelLines), variants.slice(maxModelLines)]
          : [variants, []];
-   const { isOpen, getDisclosureProps, getButtonProps, onToggle } =
-      useDisclosure();
+   const [detailsRef, isOpen, toggle] = useDetails();
    return (
-      <NextLink href={device.deviceUrl} passHref>
-         <chakra.a
-            display="flex"
-            alignItems="start"
-            transition="all 300m"
-            {...otherProps}
+      <Box
+         display="flex"
+         alignItems="start"
+         transition="all 300m"
+         {...(!truncateModels && {
+            onClick: (event) => {
+               toggle();
+               event.preventDefault();
+               event.stopPropagation();
+            },
+            onKeyDown: (event) => {
```

Comment:
Do you know why we're handling onKeyDown here?

Reply:
My intent was to toggle the dropdown if the div was focused and the user pressed 'Enter', but it looks like you can't focus the div. Do you think it should be focusable, or just omit the onKeyDown?

Reply:
Looks like the Summary element itself will be focusable, so let's omit the onKeyDown.